pkg/volume/host_path/host_path.go

```diff
@@ -99,14 +100,22 @@ func (plugin *hostPathPlugin) GetAccessModes() []v1.PersistentVolumeAccessMode {
 	}
 }
 
-func (plugin *hostPathPlugin) NewMounter(spec *volume.Spec, pod *v1.Pod, _ volume.VolumeOptions) (volume.Mounter, error) {
+func (plugin *hostPathPlugin) NewMounter(spec *volume.Spec, pod *v1.Pod, opts volume.VolumeOptions) (volume.Mounter, error) {
 	hostPathVolumeSource, readOnly, err := getVolumeSource(spec)
 	if err != nil {
 		return nil, err
 	}
 
+	path := hostPathVolumeSource.Path
+	// A containerized kubelet would have difficulty creating directories.
+	// The implementation will likely respect the containerized flag, allowing it to be "/rootfs/" aware and
+	// thus operate as desired.
+	if opts.Containerized {
+		path = filepath.Join("/rootfs", path)
```

Comment:
Just prefixing `/rootfs` for a containerized/chrooted kubelet won't work as the kernel path resolution will be off: absolute symlinks will be pointing to wrong paths (relative ones may too, in some long-chain-of-parents case), mountpoints may exists or not depending on the kubelet/host/CRI mount-ns relationship (sharing tree) and so on.

The correct approach would be to nsenter first before performing those operations, or to change the design of this to let the CRI-runtime perform hostPath creation in the host mount-ns.